azure/services/virtualmachineimages/images.go

```diff
@@ -180,14 +180,19 @@ func (s *Service) getSKUAndVersion(ctx context.Context, location, publisher, off
 
 // getUbuntuOSVersion returns the default Ubuntu OS version for the given Kubernetes version.
 func getUbuntuOSVersion(major, minor, patch uint64) string {
-	// Default to Ubuntu 20.04 LTS, except for k8s versions which have only 18.04 reference images.
-	osVersion := "2004"
+	// Default to Ubuntu 22.04 LTS, except for k8s versions which have only 20.04 or 18.04 reference images.
+	osVersion := "2204"
 	if (major == 1 && minor == 21 && patch < 2) ||
 		(major == 1 && minor == 20 && patch < 8) ||
 		(major == 1 && minor == 19 && patch < 12) ||
 		(major == 1 && minor == 18 && patch < 20) ||
 		(major == 1 && minor < 18) {
 		osVersion = "1804"
+	} else if (major == 1 && minor == 25 && patch < 3) ||
+		(major == 1 && minor == 24 && patch < 7) ||
+		(major == 1 && minor == 23 && patch < 13) ||
+		(major == 1 && minor < 23) {
```

Comment:
Does this if statement cover v1.22? I think it'll fall under line 194, but I'm not sure how explicit this needs to be.

Reply:
`v1.22.x` definitely falls in the `else if` block. I tested it here https://go.dev/play/p/LrFDOouAVUC
I also noticed that `osVersion = "1804"` gets set only for SemVers `[v1.21.0 - v1.21.1]`, any SemVers belonging to `[v1.21.2 - 1.21.x]` would also fall in the `else if` block. I am not sure if we intend that?

Reply:
I think that's the desired behavior actually. 

This is funky code, because it's capturing something external: which versions are available in the real world as reference images from the `cncf-upstream` publisher.  For v1.21.x, we starting publishing Ubuntu 20.04 versions with v1.21.2. I added some unit tests to validate this, but it's possible I missed some case.

Reply:
That makes sense. Would we also want to add a unit test case for `v1.22.x`?